tests/robot-tests/tests/admin_and_public/bau/prerelease_visibility.robot

```diff
@@ -28,13 +28,15 @@ Upload subject
     [Tags]  HappyPath
     user clicks link  Data and files
     user waits until h2 is visible  Add data file to release
+    user waits until page contains element  id:dataFileUploadForm-subjectTitle
     user enters text into element  id:dataFileUploadForm-subjectTitle   UI test subject
     user chooses file   id:dataFileUploadForm-dataFile       ${FILES_DIR}upload-file-test.csv
     user chooses file   id:dataFileUploadForm-metadataFile   ${FILES_DIR}upload-file-test.meta.csv
     user clicks button  Upload data files
 
     user waits until h2 is visible  Uploaded data files
-    user waits until page contains accordion section   UI test subject
+    user waits until element contains  id:dataFileUploadForm-subjectTitle   ${EMPTY}
+    user waits until page contains accordion section   UI test subject  60
```

Comment:
I investigated the failure on this line - it was quite hard to pin down. Looking at the screenshot of the failure from the pipeline, it was a "Sorry, there is a problem with the service" page. I tried to mess with importing subjects manually to see if I could recreate the error, but couldn't, so I assume it was happening due to the speed of the UI tests, and so I've not submitted a bug for the issue.

When trying to pin down what was happening, I tried adding they "capture large screenshot" keyword to see at what point the error was occurring, but then predictably the test case started passing.

I speculatively tried waiting on a few things - eventually trying waiting on the Subject name field being empty - and the test case now seems to pass. So it is solved but still a bit of a mystery exactly what happened here.

Reply:
hmm that's interesting. I didn't realise screenshots were available to view on the pipeline. That'll make debugging a lot easier in the future.